yourgov/components/Staff/AccessRequestsTable/AccessRequestsTable.tsx

```diff
@@ -0,0 +1,343 @@
+import { useStaffGlobalState } from '../StaffProvider';
+import { StaffMemberAccessRequest } from '../lib/accessRequests';
+import { ModalConfirmApproveAccess } from './ModalConfirmApproveAccess';
+import { ModalConfirmRejectAccess } from './ModalConfirmRejectAccess';
+import { ModalUnavailableFeature } from './ModalUnavailableFeature';
+
+export const AccessRequestsTable = () => {
+	const { accessRequestsDelete, accessRequestsGetState, accessRequestsUpdate } =
+		useStaffGlobalState();
+	const [rowSelections, setRowSelections] = useState<
+		StaffMemberAccessRequest['id'][]
+	>([]);
+	const [itemWithActiveAction, setItemWithActiveAction] =
+		useState<StaffMemberAccessRequest>();
+
+	const [approveModalOpen, setApproveModalOpen] = useState(false);
+	const [rejectModalOpen, setRejectModalOpen] = useState(false);
+	const [unavailableModalOpen, setUnavailableModalOpen] = useState(false);
+
+	const [successMessageContent, setSuccessMessageContent] = useState<
+		StaffMemberAccessRequest | StaffMemberAccessRequest[]
+	>();
+	const [successMessageType, setSuccessMessageType] = useState<
+		'approved' | 'rejected'
+	>();
+
+	const accessRequests = accessRequestsGetState();
+
+	const staffIdToStaffMemberMap = useMemo(
+		() =>
+			accessRequests.reduce<
+				Record<StaffMemberAccessRequest['id'], StaffMemberAccessRequest>
+			>(
+				(acc, staffMember) =>
+					staffMember?.firstName
+						? {
+								...acc,
+								[staffMember.id]: staffMember,
+						  }
+						: acc,
+				{}
+			),
+		[accessRequests]
+	);
+
+	const items = rowSelections.map((staffMemberId) => {
+		return staffIdToStaffMemberMap[staffMemberId];
+	});
+
+	const clearRowSelections = () => setRowSelections([]);
+
+	const onConfirmApprove = () => {
+		setSuccessMessageType('approved');
+		setSuccessMessageContent(itemWithActiveAction || items);
+		clearRowSelections();
+		accessRequestsUpdate({
+			accessRequestsToUpdate: itemWithActiveAction || items,
+			updates: {
+				status: 'Active',
+			},
+		});
+		setItemWithActiveAction(undefined);
+		setApproveModalOpen(false);
+	};
+
+	const onConfirmReject = () => {
+		setSuccessMessageType('rejected');
+		setSuccessMessageContent(itemWithActiveAction || items);
+		clearRowSelections();
+		accessRequestsDelete(itemWithActiveAction || items);
+		setItemWithActiveAction(undefined);
+		setRejectModalOpen(false);
+	};
+
+	const isSelectAllCheckboxChecked =
+		rowSelections.length === accessRequests.length;
+	const isSelectAllCheckboxIndeterminate =
+		rowSelections.length > 0 && !isSelectAllCheckboxChecked;
+	const onChangeSelectAllRows = () => {
+		if (isSelectAllCheckboxChecked) {
+			clearRowSelections();
+		} else {
+			setRowSelections(accessRequests.map(({ id }) => id));
+		}
+	};
+
+	return (
+		<>
+			{successMessageContent && successMessageType && (
+				<SectionAlert
+					focusOnUpdate={[itemWithActiveAction || items]}
+					onClose={() => {
+						setSuccessMessageContent(undefined);
+					}}
+					title={`${
```

Comment:
I wonder if the we should do something like "The access request for <name> has been <type>" so we don't have to worry about ownership issues? 
<img width="530" alt="image" src="https://github.com/user-attachments/assets/b714468f-1ede-4533-bb8e-7bee745bcafc">